### ISSUE TITLE
`brew cleanup` fails with DigitalGuardian

### ISSUE THREAD

**Issue by jak119** [Homebrew/homebrew-core#82145]
#### `brew config`

```shell
HOMEBREW_VERSION: 3.2.5
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3e52e8b48ab036af77a582219ea250107c84432f
Last commit: 3 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: a31c945bbcab1b42f530e396eaaeb2ef2b69a9dc
Core tap last commit: 15 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 12.0.5 build 1205
Git: 2.32.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.5-x86_64
CLT: 12.5.1.0.1.1623191612
Xcode: N/A
```

#### `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libnnz11.dylib
  /usr/local/lib/libociei.dylib
  /usr/local/lib/libocijdbc11.dylib
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Run `brew cleanup` without an error after it was automatically started during a routine `brew install`

### What happened (include all command output)?

```text
brew cleanup
Error: Operation not permitted @ dir_s_rmdir - /usr/local/lib/dgagent
```

### What did you expect to happen?

No errors

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew cleanup
```

Note that this is on a corporate machine so the presence of DigitalGuardian (`dgagent`) is beyond my control and other fixes such as [this one from 45009](https://github.com/Homebrew/homebrew-core/issues/45009#issuecomment-582220201) don't work because the agent stops me from modifying permissions on the folder.

**Comment by carlocab** [comment]
This was fixed in Homebrew/brew#11786. You should get the fix in the next `brew` release tag in a few days. In the meantime, see Homebrew/brew#11783 for a workaround.

Alternatively, you can use `brew developer` to switch to the development branch of `brew` until then. See `brew help developer` for details.

**Comment by jak119** [comment]
Worst part is I even scrolled through the open issues quickly and missed that. Glad to see a fix is on the way, thanks!
